src/lib.rs

```diff
@@ -0,0 +1,292 @@
+            scan_env_vars(container.env, &secret_scanner, &container.name)
+        {
+            findings.extend(secrets_in_container);
+        }
+    }
+
+    if let Some(init_containers) = pod_spec.init_containers {
+        for container in init_containers {
+            if let Some(secrets_in_container) =
+                scan_env_vars(container.env, &secret_scanner, &container.name)
+            {
+                findings.extend(secrets_in_container);
+            }
+        }
+    }
+
+    if let Some(ephemeral_containers) = pod_spec.ephemeral_containers {
+        for container in ephemeral_containers {
+            if let Some(secrets_in_container) =
+                scan_env_vars(container.env, &secret_scanner, &container.name)
+            {
+                findings.extend(secrets_in_container);
+            }
+        }
+    }
+
+    if !findings.is_empty() {
+        return kubewarden::reject_request(
+            Some(format!(
+                "The following secrets were found in environment variables -> {}",
+                create_error_message(findings)
+            )),
+            None,
+            None,
+            None,
+        );
+    }
+
+    kubewarden::accept_request()
+}
+
+fn scan_env_vars(
+    env_vars: Option<Vec<EnvVar>>,
+    secret_scanner: &SecretScanner,
+    container_name: &str,
+) -> Option<HashSet<SecretRejected>> {
+    let mut findings: HashSet<SecretRejected> = HashSet::new();
+
+    if let Some(env_vars) = env_vars {
+        for env_var in env_vars {
+            if let Some(value) = env_var.value {
+                findings.extend(scan_env_var(
+                    value.as_bytes().to_vec(),
+                    secret_scanner,
+                    env_var.name,
+                    container_name,
+                ));
+            }
+        }
+    }
+
+    if findings.is_empty() {
+        None
+    } else {
+        Some(findings)
+    }
+}
+
+fn scan_env_var(
+    input: Vec<u8>,
+    ss: &SecretScanner,
+    key: String,
+    container: &str,
+) -> HashSet<SecretRejected> {
+    let mut findings: HashSet<SecretRejected> = HashSet::new();
+    let lines = input.split(|&x| (x as char) == '\n');
+
+    for (_, new_line) in lines.enumerate() {
+        let results = ss.matches(new_line);
+        for (r, matches) in results {
+            let mut strings_found: Vec<String> = Vec::new();
+            for m in matches {
+                let result = ASCII
+                    .decode(&new_line[m.start()..m.end()], DecoderTrap::Ignore)
+                    .unwrap_or_else(|_| "<STRING DECODE ERROR>".parse().unwrap());
+                strings_found.push(result);
+            }
+            if !strings_found.is_empty() {
+                findings.insert(SecretRejected {
+                    reason: r.to_string(),
+                    key: key.clone(),
+                    container: container.to_string(),
+                });
+            }
+        }
+    }
```

Comment:
I was proposing to change this section of the code too. I put some code that took heavy inspiration from the docs of rusty-hog.